File: 백준/Gold/2437. 저울/저울.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, compare int
	fmt.Fscanf(reader, "%d\n", &n)
	var nums = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &nums[i])
	}
	compare = 1
	sort.Ints(nums)
	for _, v := range nums {
		if compare < v {
			break
		}
		compare += v
	}
	fmt.Fprintf(writer, "%d\n", compare)
}

File: 백준/Gold/2437. 저울/저울.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
nums = list(map(int, input().split()))
nums = sorted(nums)
compare = 1
for v in nums:
    if compare<v:
        break
    compare+=v
print(compare)
